artifact/tar/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+
+add_library(common_tar STATIC)
+find_package(LibArchive REQUIRED)
+if (NOT LibArchive_FOUND)
+  message (FATAL_ERROR "Could not find libarchive. Please add it to your build")
+endif()
+
+target_link_libraries(common_tar PUBLIC
+  LibArchive::LibArchive
+  common_error
+  common_log
+  common_io
+)
+target_link_libraries(common_tar PRIVATE platform_compiler_flags)
+target_link_libraries(common_tar INTERFACE crossplatform_compiler_flags)
```

Comment:
This has been changed in the main branch now. Use [this style](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/common/CMakeLists.txt#L25) instead. Cross platform flags can be removed completely.